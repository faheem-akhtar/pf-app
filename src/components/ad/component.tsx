import { FunctionComponent } from 'react';

import styles from './ad.module.scss';

export const AdComponent: FunctionComponent<{ id: string }> = ({ id }) => <div data-ad className={styles.ad} id={id} />;
