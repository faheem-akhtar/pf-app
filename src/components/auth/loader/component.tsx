import { domClassMerge } from 'helpers/dom/class-merge';

import styles from './auth-loader.module.scss';

export const AuthLoaderComponent = ({
  isEnabled,
  isCentered,
}: {
  isEnabled: boolean;
  isCentered: boolean;
}): JSX.Element | null =>
  isEnabled ? (
    <div className={styles.loader1}>
      <div
        className={domClassMerge(styles['loader1__spinner'], isCentered ? styles['loader1__spinner--centered'] : '')}
      >
        <div className={domClassMerge(styles['loader1__bounce'], styles['loader1__bounce--1'])} />
        <div className={domClassMerge(styles['loader1__bounce'], styles['loader1__bounce--2'])} />
        <div className={domClassMerge(styles['loader1__bounce'], styles['loader1__bounce--3'])} />
      </div>
    </div>
  ) : null;
