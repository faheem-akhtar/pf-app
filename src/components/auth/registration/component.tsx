import { useState } from 'react';
import { useTranslationHook } from 'helpers/hook/translation.hook';

import { ApiFetcherResultFailureInterface } from 'api/fetcher-result-failure.interface';
import { AuthRegisterService } from 'services/auth/register.service';
import { AuthRegistrationFieldEnum } from 'components/auth/registration/field.enum';
import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';
import { CheckboxTemplate } from 'library/checkbox/template';
import { formMakeValidator } from 'components/form/make-validator';
import { GoogleRecaptchaService } from 'services/google-recaptcha/service';
import { validationEmail } from 'helpers/validation/email';
import { validationRequired } from 'helpers/validation/required';

const captchaService = new GoogleRecaptchaService();

export const AuthRegistrationComponent = ({ close }: { close: () => void }): JSX.Element => {
  const { t } = useTranslationHook();

  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [firstName, setFirstName] = useState('');
  const [lastName, setLastName] = useState('');
  const [optedIn, setOptedIn] = useState(false);
  const [errors, setErrors] = useState<Partial<Record<AuthRegistrationFieldEnum, string>>>({});
  const [errorMessage, setErrorMessage] = useState('');
  const [validators] = useState(() => ({
    [AuthRegistrationFieldEnum.email]: [
      validationRequired(t('Enter your email')),
      validationEmail(t('Please enter a valid email')),
    ],
    [AuthRegistrationFieldEnum.password]: [validationRequired(t('Please enter a password'))],
    [AuthRegistrationFieldEnum.firstName]: [validationRequired(t('Please enter a valid first and last name'))],
    [AuthRegistrationFieldEnum.lastName]: [validationRequired(t('Please enter a valid first and last name'))],
  }));

  /**
   * Validate form fields
   */
  const validate = formMakeValidator(setErrors, validators);

  // TODO-FE[CX-219] - Update template
  return (
    <>
      {errorMessage && <div className='authentication__error--general'>{errorMessage}</div>}
      <form
        className='authentication__form'
        onSubmit={(e): void => {
          e.preventDefault();
          if (
            validate({
              [AuthRegistrationFieldEnum.email]: email,
              [AuthRegistrationFieldEnum.password]: password,
              [AuthRegistrationFieldEnum.firstName]: firstName,
              [AuthRegistrationFieldEnum.lastName]: lastName,
            })
          ) {
            captchaService.execute().then((captcha_token) => {
              AuthRegisterService({
                email,
                password,
                first_name: firstName,
                last_name: lastName,
                opted_in: optedIn,
                captcha_token,
              }).then((e) => {
                if (!(e as ApiFetcherResultFailureInterface).ok) {
                  setErrorMessage(
                    (e as ApiFetcherResultFailureInterface).error.body ||
                      t('Something went wrong! Please try again later')
                  );
                  captchaService.reset();
                } else {
                  // Reset server error message
                  setErrorMessage('');
                  // Close modal
                  close();
                }
              });
            });
          }
          return;
        }}
      >
        <div className='authentication__input-area'>
          <div className='authentication__input-label'>{'Enter your email'}</div>
          <input
            type='text'
            className='authentication__input'
            value={email}
            onChange={(e): void => {
              const value = e.target.value;
              validate({
                [AuthRegistrationFieldEnum.email]: value,
              });
              setEmail(value);
            }}
          />
          <div className='authentication__error'>{errors[AuthRegistrationFieldEnum.email]}</div>
        </div>
        <div className='authentication__input-area'>
          <div className='authentication__input-label'>{'Enter your first name'}</div>
          <input
            type='text'
            className='authentication__input'
            value={email}
            onChange={(e): void => {
              const value = e.target.value;
              validate({
                [AuthRegistrationFieldEnum.firstName]: value,
              });
              setFirstName(value);
            }}
          />
          <div className='authentication__error'>{errors[AuthRegistrationFieldEnum.firstName]}</div>
        </div>
        <div className='authentication__input-area'>
          <div className='authentication__input-label'>{'Enter your last name'}</div>
          <input
            type='text'
            className='authentication__input'
            value={email}
            onChange={(e): void => {
              const value = e.target.value;
              validate({
                [AuthRegistrationFieldEnum.lastName]: value,
              });
              setLastName(value);
            }}
          />
          <div className='authentication__error'>{errors[AuthRegistrationFieldEnum.lastName]}</div>
        </div>
        <div className='authentication__input-area'>
          <input
            type='password'
            className='authentication__input'
            value={password}
            onChange={(e): void => {
              const value = e.target.value;
              !validate({
                [AuthRegistrationFieldEnum.password]: value,
              });
              setPassword(value);
            }}
          />
          <div className='authentication__error'>{errors[AuthRegistrationFieldEnum.password]}</div>
        </div>
        <div className='authentication__input-area'>
          <CheckboxTemplate
            id='Opten in'
            checked={optedIn}
            onChange={(e): void => {
              setOptedIn(!!e.target.value);
            }}
          >
            {t('filters-modal/commercial-only')}
          </CheckboxTemplate>
        </div>

        <div className='authentication__form-submit'>
          <ButtonTemplate type='submit' componentType={ButtonComponentTypeEnum.tertiary} size={ButtonSizeEnum.small}>
            {'Register'}
          </ButtonTemplate>
        </div>
      </form>
    </>
  );
};
