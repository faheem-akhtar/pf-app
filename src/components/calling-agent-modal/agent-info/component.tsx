import Image from 'next/image';
import { FunctionComponent } from 'react';
import { useTranslation } from 'react-i18next';

import styles from './calling-agent-modal-agent-info.module.scss';
import { CallingAgentModalAgentInfoComponentPropsInterface } from './component-props.interface';

export const CallingAgentModalAgentInfoComponent: FunctionComponent<CallingAgentModalAgentInfoComponentPropsInterface> =
  ({ referenceId, imageSrc, languages, name }) => {
    const { t } = useTranslation();

    return (
      <div className={styles.container}>
        <div className={styles['image-container']}>
          {imageSrc ? (
            <picture>
              <source srcSet={imageSrc} type='image/jpeg' />
              <img className={styles.image} src={imageSrc} alt={name} />
            </picture>
          ) : (
            <Image
              src='/static/images/agent-image-placeholder.jpg'
              width={70}
              height={70}
              objectFit='cover'
              alt='Picture of the agent'
            />
          )}
        </div>
        <div className={styles.content}>
          <p className={styles.name}>{name}</p>
          <p className={styles.description}>
            {languages?.length ? (
              <>
                {`${t('speaks')}: ${languages.join(', ')}`}
                <br />
              </>
            ) : null}
            {t('reference')}: <span className={styles.ref}>{referenceId}</span>
          </p>
        </div>
      </div>
    );
  };
