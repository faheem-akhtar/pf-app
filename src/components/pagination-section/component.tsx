import Router, { NextRouter, useRouter } from 'next/router';
import { useContext } from 'react';

import { urlQuerySerialize } from '../../helpers/url-query/serialize';

import { ButtonSizeEnum } from 'library/button/size.enum';
import { FiltersContext } from 'components/filters/context';
import { FiltersParametersEnum } from 'enums/filters/parameters.enum';
import { LibraryButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { LibraryButtonTemplate } from 'library/button/template';
import { PaginationSectionComponentPropsType } from './component-props.type';

import styles from './pagination-section.module.scss';

// TODO-FE[TPNX-3064] Proper implementation for pagination section
Router.events.on('routeChangeStart', () => window.scrollTo({ top: 0, behavior: 'smooth' }));

const renderLink = (router: NextRouter, text: string, page: number, hidden: boolean): JSX.Element => {
  const url = `/${router.locale}${router.pathname}?${urlQuerySerialize({ ...router.query, page })}`;

  return (
    <LibraryButtonTemplate
      className={hidden ? styles.hidden : ''}
      componentType={LibraryButtonComponentTypeEnum.secondary}
      href={hidden ? '#' : url}
      size={ButtonSizeEnum.small}
      onClick={(e): void => {
        e?.preventDefault();
        router.push(url);
      }}
    >
      {text}
    </LibraryButtonTemplate>
  );
};

export const PaginationSectionComponent = ({
  pagesAvailable,
  loading,
}: PaginationSectionComponentPropsType): JSX.Element => {
  const router = useRouter();
  const {
    value: { [FiltersParametersEnum.pageNumber]: currentPage },
  } = useContext(FiltersContext);

  return (
    <div className={styles.container}>
      {renderLink(router, 'Prev', currentPage - 1, loading || currentPage === 1)}
      {renderLink(router, 'Next', currentPage + 1, loading || currentPage === pagesAvailable)}
    </div>
  );
};
