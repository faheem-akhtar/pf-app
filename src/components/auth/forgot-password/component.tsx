import React, { useState } from 'react';

import { AuthForgotPasswordFieldEnum } from 'components/auth/forgot-password/field.enum';
import { AuthForgotPasswordPropsInterface } from 'components/auth/forgot-password/props.interface';
import { AuthLoaderComponent } from 'components/auth/loader/component';
import { AuthLoginFieldEnum } from 'components/auth/login/field.enum';
import { AuthResetPasswordService } from 'services/auth/reset-password.service';
import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';
import { domClassMerge } from 'helpers/dom/class-merge';
import { ErrorMessageComponent } from 'components/error-message/component';
import { formMakeValidator } from 'components/form/make-validator';
import { GoogleRecaptchaService } from 'services/google/recaptcha.service';
import { InputBaseComponent } from 'library/input/base/component';
import { ReCaptchaComponent } from 'components/re-captcha/component';
import { useTranslation } from 'helpers/translation/hook';
import { validationEmail } from 'helpers/validation/email';
import { validationRequired } from 'helpers/validation/required';

import styles from '../auth.module.scss';

export const AuthForgotPasswordComponent = (props: AuthForgotPasswordPropsInterface): JSX.Element => {
  const captchaService = GoogleRecaptchaService();
  const { t } = useTranslation();
  const [email, setEmail] = useState('');
  const [errors, setErrors] = useState<Partial<Record<AuthLoginFieldEnum, string>>>({});
  const [errorMessage, setErrorMessage] = useState('');
  const [validators] = useState(() => ({
    [AuthLoginFieldEnum.email]: [
      validationRequired(t('validation/empty-email')),
      validationEmail(t('validation/not-valid-email')),
    ],
  }));
  const [isLoading, setIsLoading] = useState(false);

  /**
   * Validate form fields
   */
  const validate = formMakeValidator(errors, setErrors, validators);

  // TODO-FE[CX-499] - Add success message
  return (
    <div data-testid='AuthForgotPasswordComponent'>
      <div className={styles.heading}>{t('auth/forgot-password')}?</div>
      <AuthLoaderComponent isEnabled={isLoading} isCentered={true} />
      {errorMessage && <ErrorMessageComponent message={errorMessage} />}
      <form
        onSubmit={(e): void => {
          e.preventDefault();

          if (
            !validate({
              [AuthForgotPasswordFieldEnum.email]: email,
            })
          ) {
            setIsLoading(true);
            captchaService.execute().then((captcha_token) => {
              AuthResetPasswordService({
                email,
                captcha_token,
              })
                .then((e) => {
                  if (e.ok) {
                    // Reset server error message
                    setErrorMessage('');
                    // Close modal
                    props.onClose();
                    props.onSuccess();
                  } else {
                    setErrorMessage(e.error?.body || `${t('auth/something-wrong')}! ${t('auth/try-later')}`);
                    captchaService.reset();
                  }
                })
                .finally(() => setIsLoading(false));
            });
          }
          return;
        }}
      >
        <div className={styles['input-area']}>
          <InputBaseComponent
            placeholder={t('email')}
            type='email'
            value={email}
            error={!!errors[AuthForgotPasswordFieldEnum.email]}
            onChange={(value): void => {
              validate({
                [AuthForgotPasswordFieldEnum.email]: value,
              });
              setEmail(value);
            }}
            errorText={errors[AuthForgotPasswordFieldEnum.email]}
          />
        </div>
        <div className={styles['input-area']}>
          <ButtonTemplate
            className={styles.button}
            type='submit'
            componentType={ButtonComponentTypeEnum.primary}
            size={ButtonSizeEnum.regular}
          >
            {t('auth/reset-password')}
          </ButtonTemplate>
        </div>
        <ReCaptchaComponent />
      </form>
      <div className={domClassMerge(styles.link, styles['create-account'])} onClick={props.onRegister}>
        {t('auth/not-registered-yet')}?&nbsp;{t('auth/create-account')}
      </div>
    </div>
  );
};
