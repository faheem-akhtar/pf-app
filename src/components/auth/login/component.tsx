import React, { useState } from 'react';

import { AuthFacebookService } from 'services/auth/facebook.service';
import { AuthGoogleService } from 'services/auth/google.service';
import { AuthLoaderComponent } from 'components/auth/loader/component';
import { AuthLoginFieldEnum } from 'components/auth/login/field.enum';
import { AuthLoginPropsInterface } from 'components/auth/login/props.interface';
import { AuthLoginService } from 'services/auth/login.service';
import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonIconPositionEnum } from 'library/button/icon-position.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';
import { domClassMerge } from 'helpers/dom/class-merge';
import { ErrorMessageComponent } from 'components/error-message/component';
import { formMakeValidator } from 'components/form/make-validator';
import { GoogleRecaptchaService } from 'services/google-recaptcha/service';
import { IconFacebookTemplate } from 'components/icon/facebook-template';
import { IconGoogleTemplate } from 'components/icon/google-template';
import { InputBaseComponent } from 'library/input/base/component';
import { ReCaptchaComponent } from 'components/re-captcha/component';
import { useTranslation } from 'helpers/translation/hook';
import { validationEmail } from 'helpers/validation/email';
import { validationRequired } from 'helpers/validation/required';

import styles from '../auth.module.scss';
const captchaService = new GoogleRecaptchaService();

export const AuthLoginComponent = (props: AuthLoginPropsInterface): JSX.Element => {
  const { t } = useTranslation();

  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [errors, setErrors] = useState<Partial<Record<AuthLoginFieldEnum, string>>>({});
  const [errorMessage, setErrorMessage] = useState('');
  const [validators] = useState(() => ({
    [AuthLoginFieldEnum.email]: [
      validationRequired(t('validation/empty-email')),
      validationEmail(t('validation/not-valid-email')),
    ],
    [AuthLoginFieldEnum.password]: [validationRequired(t('auth/empty-password'))],
  }));
  const [isLoading, setIsLoading] = useState(false);

  /**
   * Validate form fields
   */
  const validate = formMakeValidator(errors, setErrors, validators);

  return (
    <div>
      <div className={styles.heading}>{t('sign-in')}</div>
      {/* Facebook */}
      <ButtonTemplate
        type='button'
        className={domClassMerge(styles.button, styles.facebook)}
        componentType={ButtonComponentTypeEnum.tertiary}
        size={ButtonSizeEnum.small}
        onClick={(): void => {
          setIsLoading(true);
          AuthFacebookService.signIn()
            .then(props.onClose)
            .catch(() => {
              setErrorMessage(`${t('auth/something-wrong')}! ${t('auth/try-later')}`);
              props.onClose();
            })
            .finally(() => setIsLoading(false));
        }}
        icon={{ component: IconFacebookTemplate, position: ButtonIconPositionEnum.left }}
      >
        {t('auth/sign-in-facebook')}
      </ButtonTemplate>
      {/* Google */}
      <ButtonTemplate
        type='button'
        className={domClassMerge(styles.button, styles.google)}
        componentType={ButtonComponentTypeEnum.tertiary}
        size={ButtonSizeEnum.small}
        onClick={(): void => {
          setIsLoading(true);
          AuthGoogleService.signIn()
            .then(props.onClose)
            .catch(() => {
              setErrorMessage(`${t('auth/something-wrong')}! ${t('auth/try-later')}`);
              props.onClose();
            })
            .finally(() => setIsLoading(false));
        }}
        icon={{ component: IconGoogleTemplate, position: ButtonIconPositionEnum.left }}
      >
        {t('auth/sign-in-google')}
      </ButtonTemplate>
      <div className={styles.divider}>
        <span className={styles.text}>{t('auth/or')}</span>
      </div>
      <AuthLoaderComponent isEnabled={isLoading} isCentered={true} />
      {errorMessage && <ErrorMessageComponent message={errorMessage} />}
      <form
        onSubmit={(e): void => {
          e.preventDefault();

          if (
            !validate({
              [AuthLoginFieldEnum.email]: email,
              [AuthLoginFieldEnum.password]: password,
            })
          ) {
            setIsLoading(true);
            captchaService.execute().then((captcha_token) => {
              AuthLoginService({
                email,
                password,
                captcha_token,
              })
                .then((e) => {
                  if (e.ok) {
                    // Reset server error message
                    setErrorMessage('');
                    // Close modal
                    props.onClose();
                  } else {
                    setErrorMessage(e.error.body || `${t('auth/something-wrong')}! ${t('auth/try-later')}`);
                    captchaService.reset();
                  }
                })
                .finally(() => setIsLoading(false));
            });
          }
          return;
        }}
      >
        <div className={styles['input-area']}>
          <InputBaseComponent
            placeholder={t('email')}
            type='email'
            value={email}
            error={!!errors[AuthLoginFieldEnum.email]}
            onChange={(value): void => {
              validate({
                [AuthLoginFieldEnum.email]: value,
              });
              setEmail(value);
            }}
            errorText={errors[AuthLoginFieldEnum.email]}
          />
        </div>
        <div className={styles['input-area']}>
          <InputBaseComponent
            placeholder={t('password')}
            type='password'
            value={password}
            error={!!errors[AuthLoginFieldEnum.password]}
            onChange={(value): void => {
              !validate({
                [AuthLoginFieldEnum.password]: value,
              });
              setPassword(value);
            }}
            errorText={errors[AuthLoginFieldEnum.password]}
          />
        </div>
        <div className={styles['input-area']} onClick={props.onForgotPassword}>
          <div className={domClassMerge(styles.link, styles['forgot-password'])}>{t('auth/forgot-password')}?</div>
        </div>
        <div className={styles['input-area']}>
          <ButtonTemplate
            className={styles.button}
            type='submit'
            componentType={ButtonComponentTypeEnum.primary}
            size={ButtonSizeEnum.regular}
          >
            {t('log-in')}
          </ButtonTemplate>
        </div>
        <ReCaptchaComponent />
      </form>
      <div className={domClassMerge(styles.link, styles['create-account'])} onClick={props.onRegister}>
        {t('auth/not-registered-yet')}?&nbsp;{t('auth/create-account')}
      </div>
    </div>
  );
};
