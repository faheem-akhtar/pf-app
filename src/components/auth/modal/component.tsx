import React, { useState } from 'react';

import { AuthForgotPasswordComponent } from 'components/auth/forgot-password/component';
import { AuthLoginComponent } from 'components/auth/login/component';
import { AuthRegistrationComponent } from 'components/auth/registration/component';
import { AuthScreenEnum } from 'components/auth/screen.enum';
import { IconThickCrossTemplate } from 'components/icon/thick/cross-template';
import { functionNoop } from 'helpers/function/noop';

import styles from './auth-modal-component.module.scss';
import { AuthModalPropsInterface } from './props.interface';

export const AuthModalComponent = ({
  close = functionNoop,
  cancel = functionNoop,
  success = functionNoop,
}: AuthModalPropsInterface): JSX.Element => {
  const [authScreen, setAuthScreen] = useState(AuthScreenEnum.login);
  return (
    <div
      className={styles.content}
      onClick={(e): void => {
        e.stopPropagation();
      }}
    >
      <div className={styles.header}>
        <button
          onClick={(): void => {
            close();
            cancel();
          }}
        >
          <IconThickCrossTemplate class={styles.closeIcon} />
        </button>
      </div>
      {authScreen === AuthScreenEnum.login && (
        <AuthLoginComponent
          onClose={close}
          onSuccess={success}
          onRegister={(): void => setAuthScreen(AuthScreenEnum.registration)}
          onForgotPassword={(): void => setAuthScreen(AuthScreenEnum.forgotPassword)}
        />
      )}
      {authScreen === AuthScreenEnum.registration && (
        <AuthRegistrationComponent
          onClose={close}
          onSuccess={success}
          onLogin={(): void => setAuthScreen(AuthScreenEnum.login)}
        />
      )}
      {authScreen === AuthScreenEnum.forgotPassword && (
        <AuthForgotPasswordComponent
          onClose={close}
          onSuccess={success}
          onRegister={(): void => setAuthScreen(AuthScreenEnum.registration)}
        />
      )}
    </div>
  );
};
