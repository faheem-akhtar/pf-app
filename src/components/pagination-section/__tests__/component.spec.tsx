import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { useRouter } from 'next/router';

import { mockReactUseEffect } from 'mocks/react/use-effect.mock';
import { filtersContextPropsStub } from 'stubs/filters/context-props.stub';

import { FiltersContextProvider } from 'components/filters/context-provider';

import { PaginationSectionComponent } from '../component';
import { PaginationSectionComponentPropsType } from '../component-props.type';

describe('PaginationSectionComponent', () => {
  let props: PaginationSectionComponentPropsType;

  beforeEach(() => {
    props = {
      loading: false,
      pagesAvailable: 1815,
    };
  });

  it('renders without throwing any errors', () => {
    const { container } = render(
      <FiltersContextProvider {...filtersContextPropsStub()}>
        <PaginationSectionComponent {...props} />
      </FiltersContextProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should be disable the previous button when the page is initial', () => {
    render(
      <FiltersContextProvider {...filtersContextPropsStub()}>
        <PaginationSectionComponent {...props} />
      </FiltersContextProvider>
    );

    const prevButton = screen.getByRole('link', { name: 'prev' });

    expect(prevButton).toHaveClass('disabled');
    expect(prevButton).toHaveAttribute('href', '#');
  });

  it('should call on click when the next button is clicked', () => {
    const routerPushSpy = jest.fn();
    useRouter().push = routerPushSpy;

    render(
      <FiltersContextProvider {...filtersContextPropsStub()}>
        <PaginationSectionComponent {...props} />
      </FiltersContextProvider>
    );

    const nextButton = screen.getByRole('link', { name: 'next' });

    userEvent.click(nextButton);

    expect(routerPushSpy).toHaveBeenCalledTimes(1);
    expect(routerPushSpy).toHaveBeenCalledWith('/en/search?page=2');
  });

  it('should call router events on when on mount', () => {
    mockReactUseEffect();
    window.scrollTo = jest.fn();

    const handlers: Function[] = [];
    useRouter().events.on = (_, callback): void => {
      handlers.push(callback);
    };

    render(
      <FiltersContextProvider {...filtersContextPropsStub()}>
        <PaginationSectionComponent {...props} />
      </FiltersContextProvider>
    );

    handlers.forEach((handler) => handler(''));

    expect(window.scrollTo).toHaveBeenCalledWith({ behavior: 'smooth', top: 0 });
  });

  it('should call router events off when on unmount', () => {
    const { unmountAll } = mockReactUseEffect();
    window.scrollTo = jest.fn();

    const routerOffSpy = jest.fn();
    useRouter().events.off = routerOffSpy;

    render(
      <FiltersContextProvider {...filtersContextPropsStub()}>
        <PaginationSectionComponent {...props} />
      </FiltersContextProvider>
    );

    unmountAll();

    expect(routerOffSpy).toHaveBeenCalledWith('routeChangeStart', expect.anything());
  });
});
