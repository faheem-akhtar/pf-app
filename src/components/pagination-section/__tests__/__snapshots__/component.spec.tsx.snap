// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationSectionComponent renders without throwing any errors 1`] = `
<div>
  <div
    class="container"
  >
    <a
      class="button primary small disabled primary--disabled button"
      disabled=""
      href="#"
    >
      <svg
        class="icon icon--left leftIcon"
        height="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          clip-rule="evenodd"
          d="M13.588 19.2929C13.9785 19.6834 14.6117 19.6834 15.0022 19.2929C15.3911 18.904 15.393 18.2741 15.0065 17.8829L9.17257 11.9782L15.003 6.11596C15.391 5.72576 15.3902 5.0951 15.001 4.70596C14.6111 4.31607 13.979 4.31607 13.5891 4.70596L7.00218 11.2929C6.61166 11.6834 6.61166 12.3166 7.00218 12.7071L13.588 19.2929Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      prev
    </a>
    <a
      class="button primary small button"
      href="/en/search?page=2"
    >
      next
      <svg
        class="icon icon--right rightIcon"
        height="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          clip-rule="evenodd"
          d="M10.4121 4.70711C10.0216 4.31658 9.38844 4.3166 8.99792 4.70713C8.60906 5.09599 8.60714 5.72588 8.99365 6.11708L14.8275 12.0218L8.99716 17.884C8.60908 18.2742 8.60994 18.9049 8.99908 19.294C9.38897 19.6839 10.0211 19.6839 10.411 19.294L16.9979 12.7071C17.3885 12.3166 17.3885 11.6834 16.9979 11.2929L10.4121 4.70711Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </a>
  </div>
</div>
`;
